src/frontend/configuration/node_migration.rs

```diff
@@ -0,0 +1,563 @@
+        match self {
+            MigrationMode::Migrate => T.node_migration_mode_migrate_explanation().to_string(),
+            MigrationMode::FreshSync => T.node_migration_mode_fresh_sync_explanation().to_string(),
+            MigrationMode::ResetInPlace => T.node_migration_mode_reset_explanation().to_string(),
+        }
+    }
+
+    /// Returns the button label for starting this mode
+    fn start_button_label(&self) -> String {
+        match self {
+            MigrationMode::Migrate | MigrationMode::FreshSync => {
+                T.node_migration_button_start().to_string()
+            }
+            MigrationMode::ResetInPlace => T.node_migration_button_reset().to_string(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum NodeMigrationInput {
+    OpenDestinationDialog,
+    DestinationSelected(PathBuf),
+    MigrationModeChanged(MigrationMode),
+    StartMigration,
+    Cancel,
+    Ignore,
+    /// Update source size after async calculation
+    SourceSizeCalculated(u64),
+    /// Update destination free space after async calculation
+    DestinationSpaceCalculated(u64),
+    /// Update whether non-node data was detected in source
+    NonNodeDataDetected(bool),
+}
+
+#[derive(Debug, Clone)]
+pub enum NodeMigrationOutput {
+    StartMigration {
+        source: PathBuf,
+        destination: PathBuf,
+        snap_sync: bool,
+    },
+    Cancel,
+}
+
+pub struct NodeMigrationInit {
+    pub source_path: PathBuf,
+    pub parent_window: gtk::Window,
+    /// Initial migration mode to pre-select
+    pub initial_mode: MigrationMode,
+}
+
+#[tracker::track]
+#[derive(Debug)]
+pub struct NodeMigrationDialog {
+    source_path: PathBuf,
+    source_size: Option<u64>,
+    destination_path: Option<PathBuf>,
+    destination_free_space: Option<u64>,
+    destination_valid: bool,
+    migration_mode: MigrationMode,
+    /// True if non-node data was detected in the source directory
+    has_non_node_data: bool,
+    #[do_not_track]
+    open_dialog: Controller<OpenDialog>,
+}
+
+impl NodeMigrationDialog {
+    /// Returns the minimum required space at destination for migration
+    fn required_destination_space(&self) -> Option<u64> {
+        self.source_size
+    }
+
+    /// Check if destination has enough space for migration
+    fn has_sufficient_space(&self) -> bool {
+        if self.migration_mode.uses_snap_sync() {
+            // Snap sync only needs NODE_FREE_SPACE_WARNING_THRESHOLD
+            if let Some(dest_space) = self.destination_free_space {
+                return dest_space >= NODE_FREE_SPACE_WARNING_THRESHOLD;
+            }
+            return false;
+        }
+
+        if let (Some(required), Some(dest_space)) =
+            (self.required_destination_space(), self.destination_free_space)
+        {
+            dest_space >= required
+        } else {
+            false
+        }
+    }
+
+    /// Returns true if destination is fully valid (exists, writable, and has enough space)
```

Comment:
writable part of the comment seems missing in the code. Or is the write check implicit somewhere  else ?